YololCompetition/Services/Messages/DBMessages.cs

```diff
@@ -0,0 +1,106 @@
+using Discord;
+
+namespace YololCompetition.Services.Messages
+{
+    public class DbMessages
+        : IMessages
+    {
+        private readonly IDatabase _database;
+        private readonly ICron _cron;
+        private readonly IChallenges _challenges;
+        private readonly BaseSocketClient _client;
+
+        public enum messageType : uint {Current, Leaderboard}
+        
+        public DbMessages(IDatabase database, ICron cron, IChallenges challenges, BaseSocketClient client)
+        {
+            _database = database;
+            _cron = cron;
+            _challenges = challenges;
+            _client = client;
+
+            try
+            {
+                _database.Exec("CREATE TABLE IF NOT EXISTS `Messages` (`ChannelID` INTEGER NOT NULL, `MessageID` INTEGER NOT NULL, `ChallengeID` INTEGER NOT NULL, 'MessageType' INTEGER NOT NULL, PRIMARY KEY(`MessageID`, `ChallengeID'))");
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+            }
+        }
+
+        public async Task TrackMessage(ulong ChannelID, ulong MessageID, ulong ChallengeID, uint MessageType)
+        {
+            await using var cmd = _database.CreateCommand();
+            cmd.CommandText = "INSERT into Messages (ChannelID, MessageID, ChallengeID, MessageType) values(@ChannelID, @MessageID, @ChallengeID, @MessageType)";
+            cmd.Parameters.Add(new SqliteParameter("@ChannelID", DbType.UInt64) { Value = ChannelID });
+            cmd.Parameters.Add(new SqliteParameter("@MessageID", DbType.UInt64) { Value = MessageID });
+            cmd.Parameters.Add(new SqliteParameter("@ChallengeID", DbType.UInt64) { Value = ChallengeID });
+            cmd.Parameters.Add(new SqliteParameter("@MessageType", DbType.UInt64) { Value = MessageType });
+            await cmd.ExecuteNonQueryAsync();
+        }
+
+        public IAsyncEnumerable<Message> GetCurrentMessages(ulong challengeID)
+        {
+            DbCommand PrepareQuery(IDatabase database)
+            {
+                var cmd = _database.CreateCommand();
+                cmd.CommandText = "SELECT from 'Messages' WHERE 'ChallengID' = @challengeID";
+                cmd.Parameters.Add(new SqliteParameter("@ChallengeID", DbType.UInt64) { Value = challengeID});
+                return cmd;
+            }
+
+            return new SqlAsyncResult<Message>(_database, PrepareQuery, ParseMessage);
+        }
+
+        private static Message ParseMessage(DbDataReader reader)
+        {
+            return new Message(
+                ulong.Parse(reader["ChannelID"].ToString()!),
+                ulong.Parse(reader["MessageID"].ToString()!),
+                ulong.Parse(reader["ChallengeID"].ToString()!),
+                uint.Parse(reader["MessageType"].ToString()!)
+            );
+        }
+
+        public void StartMessageWatch()
+        {
+            _cron.Schedule(TimeSpan.FromMinutes(5), TimeSpan.FromMinutes(1), uint.MaxValue, async () => {
```

Comment:
This job will not update embeds after the challenge ends (to add the `Challenge has ended` text).

Reply:
Extracted Final Update to a separate method (`FinalUpdateMessages()` )that is run by the thing that ends the current competition. This updates them one last time (should get the ended message since it will be after the time the challenge ends) and removes them from the db messages table.